micro-rdk/src/esp32/entry.rs

```diff
@@ -121,15 +138,34 @@ pub async fn serve_web_inner(
     };
 
     #[cfg(feature = "data")]
-    // TODO: Spawn data task here. May have to move the initialization below to the task itself
     // TODO: Support implementers of the DataStore trait other than StaticMemoryDataStore in a way that is configurable
-    {
-        let _data_manager_svc = DataManager::<StaticMemoryDataStore>::from_robot_and_config(
-            &cfg_response,
-            &app_config,
-            robot.clone(),
-        );
-    }
+    let data_manager_svc = match DataManager::<StaticMemoryDataStore>::from_robot_and_config(
+        &cfg_response,
+        &app_config,
+        robot.clone(),
+    ) {
+        Ok(svc) => svc,
+        Err(err) => {
+            log::error!("error configuring data management: {:?}", err);
+            None
+        }
+    };
+
+    #[cfg(feature = "data")]
+    let data_sync_task = data_manager_svc
+        .as_ref()
+        .map(|data_manager_svc| data_manager_svc.get_sync_task());
+
+    #[cfg(feature = "data")]
+    let data_future = Box::pin(async move {
+        if let Some(mut data_manager_svc) = data_manager_svc {
+            if let Err(err) = data_manager_svc.data_collection_task().await {
+                log::error!("error running data manager: {:?}", err)
+            }
+        }
+    });
+    #[cfg(not(feature = "data"))]
+    let data_future = async move {};
```

Comment:
Because I believe the fix that `Box::pin` is providing is that it offloads some of the high-memory async code to the heap and away from the same stack memory generated for the outerlying async function. In this case, the function uses almost no memory and is short-lived on the stack so it doesn't need to be pinned to the heap